src/debug/telemetry/telemetry.c

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Looking for a good home for this file, and the .h one. I created telemetry folder in debug, but if you think there's a more fittting place let me know.